### PR TITLE
[BugFix] NPE for show routine load stmt

### PR DESCRIPTION
**Pull request by rickif**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes https://github.com/StarRocks/StarRocksTest/issues/1955

## Problem Summary(Required) ：

The `dbCtx` and `nameCtx` of show routine load stmt is possibly null when there's no db name/job name specified.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by rickif** [comment]
run starrocks_fe_unittest

**Review by packy92 — DISMISSED** [review]

**Review by Seaven — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java [hunk 1]
```diff
@@ -6055,19 +6055,23 @@ private NodePosition createPos(Token start, Token stop) {
...
 
+        String dbName = null;
         if (dbCtx != null) {
-            start = dbCtx.start;
             dbName = getQualifiedName(dbCtx).toString();
-
+            start = dbCtx.start;
         }
+
         return new LabelName(dbName, name, createPos(start, stop));
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java:6075]: If dbCtx and nameCtx all are NULL, what's difference between
```
return new LabelName(null, null, NodePosition.ZERO);
```
and 
```
return new LabelName(dbName, name, createPos(start, stop));
```

Comment (rickif) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java:6075]: They are the same.

**Review by packy92 — APPROVED** [review]

**Comment by rickif** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 7 / 7 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|7|7|100.00%|[]|

**Review by chaoyli — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by Seaven — APPROVED** [review]

**Comment by rickif** [comment]
run starrocks_admit_test

**Comment by rickif** [comment]
run starrocks_admit_test

**Comment by rickif** [comment]
run starrocks_admit_test

**Comment by rickif** [comment]
run starrocks_admit_test

**Comment by rickif** [comment]
run starrocks_admit_test

**Comment by rickif** [comment]
run starrocks_admit_test

**Pull request merged**